packages/uif/src/BaseIndexer.test.ts

```diff
@@ -55,15 +59,106 @@ describe(BaseIndexer.name, () => {
       await child.finishInvalidate()
 
       await parent.doTick(10)
+      await parent.finishTick(10)
       await middle.finishUpdate(10)
 
       expect(child.getState().status).toEqual('updating')
 
       await parent.doTick(5)
+      await parent.finishTick(5)
 
       expect(middle.getState().waiting).toEqual(true)
     })
   })
+
+  describe('retries on error', () => {
+    // Those tests are quite slow (around 500ms total)
+    // This is not a problem rn, but can be improved in the future
+    it('invalidates and retries update', async () => {
+      const clock = install({ shouldAdvanceTime: true })
+      const parent = new TestRootIndexer(0)
+      const updateRetryStrategy: RetryStrategy = {
+        shouldRetry: () => true,
+        timeoutMs: () => 1000,
+        clear: () => {},
+      }
+
+      const child = new TestChildIndexer([parent], 0, '', {
+        updateRetryStrategy,
+      })
+
+      await parent.start()
+      await child.start()
+
+      await child.finishInvalidate()
+
+      await parent.doTick(1)
+      await parent.finishTick(1)
+
+      await child.finishUpdate(new Error('test error'))
+      expect(child.updating).toBeFalsy()
+      expect(child.getState().status).toEqual('invalidating')
+
+      await child.finishInvalidate()
+      await clock.tickAsync(1000)
```

Comment:
add idle check here and in other tests too

Reply:
fe935f3